os/tools/symtab_formatter.c

```diff
@@ -0,0 +1,133 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <stdlib.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define ADDR_LENGTH 4
+#define HEXADECIMAL_BASE 16
+#define MAP_FILE_PATH "../build/output/bin/System.map"
+#define BIN_FILE_PATH "../build/output/bin/Symtab.bin"
+#define MAX_NUM_SYM 8000
+#define MAX_SYM_LEN 128
+
+/****************************************************************************
+ * Global Functions
+ ****************************************************************************/
+
+uint32_t symbolnum;
+char symbolname[MAX_NUM_SYM][MAX_SYM_LEN];
+uint32_t symboladdress[MAX_NUM_SYM] = {0};
+uint32_t symboloffset[MAX_NUM_SYM] = {0};
+char symbol[2 * MAX_SYM_LEN]={0};
+
+int main()
+{
+	FILE *fp;
+	int len;
+	bool space;
+	char temp_addr[MAX_SYM_LEN];
+	int cnt;
+	unsigned int addr;
+	size_t ret;
+
+	fp = fopen(MAP_FILE_PATH, "r");
+
+	/* If the file could not be opened */
+	if (fp == NULL) {
+		printf("Symtab.bin creation failed \n");
+		return -1;
+	}
+
+	/* Reading the system.map file line by line */
+	while (fgets(symbol, sizeof(symbol), fp) != NULL) {
+		len = 0;
+		space = false;
+		memset(temp_addr, '\0', sizeof(temp_addr));
+		cnt = 0;
+
+		while (len < strlen(symbol)) {
```

Comment:
I think it is possible to implement the parsing logic without the use of while loop. 
Please consider the below code to extract the symboladdress and symbolname values;

`char *ptr;

ptr = symbol;

symboladdress[symbolnum] = (unsigned int)strtoul(ptr, &ptr, HEXADECIMAL_BASE);
strcpy(symbolname[symbolnum], (char *)(ptr + 3));
symbolname[symbolnum][strlen(symbolname[symbolnum])] = '\0';
`